modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/resource/v1_0/TaxonomyCategoryResourceImpl.java

```diff
@@ -575,35 +572,48 @@ private String[] _merge(
 		List<AssetCategoryProperty> assetCategoryProperties,
 		TaxonomyCategoryProperty[] taxonomyCategoryProperties) {
 
-		Stream<TaxonomyCategoryProperty> stream = Arrays.stream(
-			Optional.ofNullable(
-				taxonomyCategoryProperties
-			).orElse(
-				new TaxonomyCategoryProperty[0]
-			));
+		if (ArrayUtil.isEmpty(taxonomyCategoryProperties)) {
+			String[] categoryProperties =
+				new String[assetCategoryProperties.size()];
 
-		Map<String, String> map = stream.collect(
-			Collectors.toMap(
-				TaxonomyCategoryProperty::getKey,
-				TaxonomyCategoryProperty::getValue));
+			for (int i = 0; i < categoryProperties.length; i++) {
+				AssetCategoryProperty assetCategoryProperty =
+					assetCategoryProperties.get(i);
 
-		for (AssetCategoryProperty assetCategoryProperty :
-				assetCategoryProperties) {
+				categoryProperties[i] =
+					assetCategoryProperty.getKey() + ":" +
+						assetCategoryProperty.getValue();
+			}
+
+			return categoryProperties;
+		}
+
+		List<String> categoryProperties = new ArrayList<>();
+		Set<String> keys = new HashSet<>();
+
+		for (TaxonomyCategoryProperty taxonomyCategoryProperty :
```

Comment:
@slnn The logic before/after may not be consistent. What if those list/array themselves has duplicate item? When using a map it's overwritten/skipped, while in your logic it will be kept.

Reply:
Oh, yeah, my change will not skip the duplicate item....

Reply:
Oh, yeah, my change will not skip the duplicate item....